docs/objects/request.md

```diff
@@ -409,3 +409,29 @@ $app->add(function ($request, $response, $next) {
     return $next($request, $response);
 });
 {% endhighlight %}
+
+## Attributes
+
+With PSR-7 it is possible to inject objects/values into the request object for further processing. In your applications middleware often need to pass along information to your route closure and the way to do is it is to add it to the request object via an attribute.
+
+Example, Setting a value on your request object.
+
+{% highlight php %}
+$app->add(function ($request, $response, $next) {
+    $request = $request->withAttribute('session', $_SESSION); //add the session storage to your request
```

Comment:
Note that `$_SESSION` is an array and you only copy its contents. This means you pass it as read only and cannot modify the session via the `session` attribute. This is fine, but it might confuse some people.